ocaml/typing/typecore.ml

```diff
@@ -8367,6 +8372,8 @@ let report_type_expected_explanation expl ppf =
       because "a range-based for iterator stop index in a comprehension"
   | Comprehension_when ->
       because "a when-clause in a comprehension"
+  | Error_message_attr msg ->
+      fprintf ppf "@ because [@error_message]: %s" msg
```

Comment:
If you agree with my suggestion not to include the `because`, then use a separator other than `@ `; some of the others force a newline (I forget which).